lightkube/core/client.py

```diff
@@ -176,6 +176,69 @@ def watch(self, res, *, namespace=None, labels=None, fields=None, server_timeout
         )
         return self._client.watch(br, on_error=on_error)
 
+    @overload
+    def wait(
+        self,
+        res: Type[GlobalResource],
+        name: str,
+        *,
+        for_conditions: Iterable[str] = (),
+        raise_for_conditions: Iterable[str] = (),
+    ) -> GlobalResource:
+        ...
+
+    @overload
+    def wait(
+        self,
+        res: Type[AllNamespacedResource],
+        name: str,
+        *,
+        namespace: str = None,
+        for_conditions: Iterable[str] = (),
+        raise_for_conditions: Iterable[str] = (),
+    ) -> AllNamespacedResource:
+        ...
+
+    def wait(
+        self,
+        res,
+        name: str,
+        *,
+        namespace=None,
+        for_conditions: Iterable[str] = (),
```

Comment:
I guess you put the parameter here and optional for consistency, but in this way your editor (or pylint) won't show an error if this parameter is not provided. I think it's fine to have it required but we can force it to be a keyword argument so that position won't matter as far as it is defined.

Reply:
Reopening this one, I was suggesting to force it to be a keyword argument (so appear after the '*' ) but right now it seems to be only positional. The reason is that if in the future we want to add `for_delete`, we will need to bring `for_conditions` back as optional (need to be mutually exclusive with `for_delete`) and we want to be sure we can do that in a backward compatibility way.

Reply:
I think after this we are good to merge 👍

Reply:
Ah sure, sounds good. Updated

Reply:
Whoops, missed a spot, as I was accidentally filtering the tests to be ran from `e2e-tests/` when running locally. Let me update that.

Reply:
Alright, updated. Sorry about that